addons/sale/models/sale_order_line.py

```diff
@@ -81,7 +81,7 @@ class SaleOrderLine(models.Model):
     product_id = fields.Many2one(
         comodel_name='product.product',
         string="Product",
-        change_default=True, ondelete='restrict', check_company=True, index='btree_not_null',
+        change_default=True, ondelete='restrict', index='btree_not_null',
```

Comment:
I think we should add the parent check in the domain just below.
Without that, I think removing this check risk to display products from a completely unrelated company.

Did you check whether that was the case or not?

Reply:
We have a constrain check for product_id company already, in
- [here](https://github.com/odoo/odoo/pull/140078/files#diff-86168e4c37e9a7c1437998990c2aa4e1690d16605812f57a81172936377b9c50R157-R168) for purchase
- [here](https://github.com/odoo/odoo/pull/140078/files#diff-695321b9df7129604a59fbc9bff34f97ba90b3394aa1166ca24e36b27389ae82R668-R679) for sale

Reply:
This is preventing to save the invoice with those product but not to display them.
But I tested in the UI and the products don't show up so it's ok! :ok_hand: